File: abstractfactory.php
<?php

// Abstract Product A
interface Chair {
    public function create();
}

// Abstract Product B
interface Table {
    public function create();
}

// Concrete Product A1
class ModernChair implements Chair {
    public function create() {
        echo "Modern Chair created.\n";
    }
}

// Concrete Product A2
class VictorianChair implements Chair {
    public function create() {
        echo "Victorian Chair created.\n";
    }
}

// Concrete Product B1
class ModernTable implements Table {
    public function create() {
        echo "Modern Table created.\n";
    }
}

// Concrete Product B2
class VictorianTable implements Table {
    public function create() {
        echo "Victorian Table created.\n";
    }
}

// Abstract Factory
interface FurnitureFactory {
    public function createChair(): Chair;
    public function createTable(): Table;
}

// Concrete Factory 1
class ModernFurnitureFactory implements FurnitureFactory {
    public function createChair(): Chair {
        return new ModernChair();
    }
    public function createTable(): Table {
        return new ModernTable();
    }
}

// Concrete Factory 2
class VictorianFurnitureFactory implements FurnitureFactory {
    public function createChair(): Chair {
        return new VictorianChair();
    }
    public function createTable(): Table {
        return new VictorianTable();
    }
}

// Client
function createFurnitureSet(FurnitureFactory $factory) {
    $chair = $factory->createChair();
    $table = $factory->createTable();

    $chair->create();
    $table->create();
}

echo "Modern Furniture Set:\n";
createFurnitureSet(new ModernFurnitureFactory());

echo "\nVictorian Furniture Set:\n";
createFurnitureSet(new VictorianFurnitureFactory());
?>

File: AbstractFactory.cs
// C# Implementation

using System;

// Abstract Product A
interface IChair {
    void Create();
}

// Abstract Product B
interface ITable {
    void Create();
}

// Concrete Product A1
class ModernChair : IChair {
    public void Create() {
        Console.WriteLine("Modern Chair created.");
    }
}

// Concrete Product A2
class VictorianChair : IChair {
    public void Create() {
        Console.WriteLine("Victorian Chair created.");
    }
}

// Concrete Product B1
class ModernTable : ITable {
    public void Create() {
        Console.WriteLine("Modern Table created.");
    }
}

// Concrete Product B2
class VictorianTable : ITable {
    public void Create() {
        Console.WriteLine("Victorian Table created.");
    }
}

// Abstract Factory
interface IFurnitureFactory {
    IChair CreateChair();
    ITable CreateTable();
}

// Concrete Factory 1
class ModernFurnitureFactory : IFurnitureFactory {
    public IChair CreateChair() {
        return new ModernChair();
    }
    public ITable CreateTable() {
        return new ModernTable();
    }
}

// Concrete Factory 2
class VictorianFurnitureFactory : IFurnitureFactory {
    public IChair CreateChair() {
        return new VictorianChair();
    }
    public ITable CreateTable() {
        return new VictorianTable();
    }
}

// Client
class FurnitureStore {
    private IChair chair;
    private ITable table;

    public FurnitureStore(IFurnitureFactory factory) {
        chair = factory.CreateChair();
        table = factory.CreateTable();
    }

    public void CreateFurniture() {
        chair.Create();
        table.Create();
    }
}

class Program {
    static void Main() {
        Console.WriteLine("Modern Furniture Set:");
        FurnitureStore modernStore = new FurnitureStore(new ModernFurnitureFactory());
        modernStore.CreateFurniture();

        Console.WriteLine("\nVictorian Furniture Set:");
        FurnitureStore victorianStore = new FurnitureStore(new VictorianFurnitureFactory());
        victorianStore.CreateFurniture();
    }
}
